pkgs/build-support/php/build-composer-deps.nix

```
@@ -0,0 +1,53 @@
+{ stdenvNoCC, lib, writeTextDir, php }:
+
+let
+  buildComposerDepsOverride = finalAttrs: previousAttrs:
```

For documentation purposes, it might be nice to include the destructing in a comment:

```suggestion
  buildComposerDepsOverride =
  /*
  We cannot destruct finalAttrs since the attrset below is used to construct it
  and Nix currently does not support lazy attribute names.
  {
    php ? null,
    composer ? null,
    composerLock ? "composer.lock",
    src,
    vendorHash
    ...
  }@finalAttrs:
  */
  finalAttrs:
  previousAttrs:
```

---

fixed.